frontend/instance/frontend/init-instance.sh

```diff
@@ -10,6 +10,19 @@ aws --region $region s3 cp --recursive s3://membership-dist/${stack}/${stage}/fr
 mkdir /etc/gu
 aws --region $region s3 cp s3://membership-private/${stage}/membership.private.conf /etc/gu
 
+# wait for dpkg lock to become free
+lsof /var/lib/dpkg/lock
+rc=$?
+echo "Return code is $rc"
+until [ "$rc" -eq "1" ]; do
+  echo "Waiting for dpkg lock to become free"
+  sleep 1s;
+  lsof /var/lib/dpkg/lock
+  rc=$?
+  echo "Return code is $rc"
+done
+echo "dpkg lock is free, ready to install application"
+
 # install and start play app and create frontend user
 dpkg -i /dist/frontend_1.0-SNAPSHOT_all.deb
```

Comment:
Obviously it is only reducing the chance to have a contention issue, as a lock could be taken between your last lock check and the call to `dpkg`, but the probability is very low.

Through this [stackoverflow answer](https://askubuntu.com/a/231859) I discovered  `aptdcon` which is another way to solve this problem, but requires it to be installed on the base image.